eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/commons/Extra.java

```
@@ -0,0 +1,40 @@
+package com.schedjoules.eventdiscovery.framework.serialization.commons;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.schedjoules.eventdiscovery.framework.serialization.Keys;
+import com.schedjoules.eventdiscovery.framework.serialization.core.Key;
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+import com.schedjoules.eventdiscovery.framework.utils.optionals.AbstractCachingOptional;
+
+import org.dmfs.optional.Absent;
+import org.dmfs.optional.Optional;
+
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public final class Extra<T> extends AbstractCachingOptional<T>
+{
+
+    public Extra(final Key<T> key, final Intent intent)
+    {
+        super(new Factory<Optional<T>>()
+        {
+            @Override
+            public Optional<T> create()
+            {
+                if (intent != null)
```

I guess this could be replaced by
```
return new Argument<T>(key, new NestedBundle(intent).value());
```

or even better, if `Argument` would take an `Optional<Bundle>` 
```
return new Argument<T>(key, new NestedBundle(intent)); 
```

---

Btw. don't check `intent` for `null`. If you pass a `null` Intent you're most likely doing it wrong and the code should not swallow this error silently.

---

Right, simplified it and also removed the `null` check for `intent` - you're right that can never really be `null` normally.